src/magic_folder/config.py

```diff
@@ -118,10 +126,60 @@
         )
         """,
         """
-        CREATE TABLE local_snapshots
+        CREATE TABLE [local_snapshots]
         (
-            path          TEXT PRIMARY KEY,  -- the (mangled) name in UTF8
-            snapshot_blob BLOB               -- a JSON blob representing the snapshot instance
+            -- A random, unique identifier for this particular snapshot.
+            [identifier]       TEXT PRIMARY KEY,
+
+            -- The magicpath-mangled name of the file this snapshot is for,
+            -- UTF-8-encoded.
+            [path]             TEXT,
+
+            -- A local filesystem path where the content can be found.
+            [content_path]     TEXT
+        )
+        """,
+        """
+        CREATE TABLE [local_snapshot_metadata]
+        (
+            -- The identifier of the snapshot which this metadata row belongs to.
+            [snapshot_identifier]   TEXT NOT NULL,
+
+            -- The metadata key for this row.
+            [key]                   TEXT NOT NULL,
+
+            -- The handy SQLite3 "BLOB" storage class allows us to store a
+            -- value of any supported type, even differing from row to row.
+            [value]                 BLOB NOT NULL,
+
+            -- The referenced local snapshot must exist.
+            FOREIGN KEY([snapshot_identifier]) REFERENCES [local_snapshot]([identifier])
+        )
+        """,
+        """
+        CREATE TABLE [local_snapshot_parent]
+        (
+            -- The identifier of the snapshot to which this parent belongs.
+            [snapshot_identifier]   TEXT NOT NULL,
+
+            -- The index of this parent in the snapshot's ordered parent list.
+            -- Offer marginal additional data integrity by requiring it to be
+            -- 0 or greater.
+            [index]                 INTEGER CHECK ([index] >= 0) NOT NULL,
+
+            -- If this parent is local only then its identifier is only
+            -- meaningful for looking up local snapshots in our local
+            -- database.  If it is not only local then its identifier is a
+            -- capability refering to a remote snapshot which we may or may
+            -- not have a local cache of.
+            [local_only]            BOOL NOT NULL,
```

Comment:
Just to capture some real-time comments from elsewhere, this table strikes me as trying to mash two things together: local and remote parents. It might be better to have them separate so there isn't this "bool" column that essentially everything using this will have to consult to figure out how to interpret the "id" column.

The actual work of splitting this could be done when we actually have remote AND local parents? Could be worth a comment here though.

Reply:
Added a comment.  I agree it makes sense to defer implementation changes here until there's application code depending on this stuff and we can see what facilitates the higher level goals better.